controllers/administration.js

```diff
@@ -2596,100 +2596,127 @@ router.delete(
 	deleteEventsForData('courses'),
 );
 
+const buildArchiveQuery = (courseStatus) => {
+	const yesterday = new Date();
+	yesterday.setDate(yesterday.getDate() - 1);
+	let untilQuery = {};
+	if (courseStatus === 'active') {
+		untilQuery = {
+			$or: [
+				{ untilDate: { $exists: false } },
+				{ untilDate: null },
+				{ untilDate: { $gte: yesterday } },
+			],
+		};
+	}
+	if (courseStatus === 'archived') {
+		untilQuery = { untilDate: { $lt: yesterday } };
+	}
+	return untilQuery;
```

Comment:
please use same names, as the function is called buildArchiveQuery, use archiveQuery instead of untilQuery above and in line 2620 again.